### PR TITLE
MonthPicker is added

### PR DESCRIPTION
**Pull request by ertugrul-aydin**
I have added the MonthPicker component. I noticed also that clear button is not working yet on the MonthPicker component. Additionally, it seems that it needs to make some modifications to the filters by including '$lte' for it to work correctly.

### CODE REVIEW

File: apps/dashboard/pages/donations.tsx [hunk 1]
```diff
@@ -7,13 +7,21 @@ import { useRouter } from 'next/router'
...
+  startYear: number
+  startMonth: number
+  endYear: number
+  endMonth: number
+}
 
 const DonationsPage: FC<PageProps> = ({ seo }) => {
   const [currentPage, setCurrentPage] = useState<number>(1)
   const [searchTerm, setSearchTerm] = useState<string>()
+  const [month, setMonth] = useState<Month>()
```
Comment (7alip) [inline apps/dashboard/pages/donations.tsx:24]: `month.startYear` seems weird. State name could be something like `date`, `range` or `dateRange`

```suggestion
  const [date, setDate] = useState<Month>()
```

File: apps/dashboard/pages/donations.tsx [hunk 2]
```diff
@@ -7,13 +7,21 @@ import { useRouter } from 'next/router'
...
+import { MonthPicker } from '@wsvvrijheid/ui'
 import { AdminLayout, DataTable, PageHeader, useColumns } from '@wsvvrijheid/ui'
 
 type PageProps = InferGetStaticPropsType<typeof getStaticProps>
+type Month = {
+  startYear: number
+  startMonth: number
+  endYear: number
+  endMonth: number
+}
```
Comment (7alip) [inline apps/dashboard/pages/donations.tsx:19]: There is already `RangeParams` from MonthPicker

File: apps/dashboard/pages/donations.tsx [hunk 3]
```diff
@@ -48,6 +70,7 @@ const DonationsPage: FC<PageProps> = ({ seo }) => {
         onSearch={handleSearch}
         searchPlaceHolder={'Search arts by title or artist'}
       />
+      <MonthPicker onRangeSelect={handleSelect} />
```
Comment (7alip) [inline apps/dashboard/pages/donations.tsx:73]: `MonthPicker` should be children of `PageHeader`

```tsx
<PageHeader>
  <MonthPicker />
</PageHeader>

Comment (7alip) [reply apps/dashboard/pages/donations.tsx:73]: It seems we're not able to use clear method outside of the component.

To fix this, you can add `onClear` prop to MonthPicker. Whenever user clicks `Clear` button, you can both clear internal state and execute `onClear` function.

**Review by 7alip — CHANGES_REQUESTED** [review]
Thank you

File: apps/dashboard/pages/donations.tsx [hunk 4]
```diff
@@ -26,6 +34,16 @@ const DonationsPage: FC<PageProps> = ({ seo }) => {
...
     filters: {
       ...(searchTerm && { email: { $containsi: searchTerm } }),
+      ...(month && {
+        createdAt: {
+          $gte: `${month?.startYear}-${month?.startMonth
+            .toString()
+            .padStart(2, '0')}-01T00:00:00.000Z`,
+          $lte: `${month?.endYear}-${month?.endMonth
+            .toString()
+            .padStart(2, '0')}-01T00:00:00.000Z`,
```
[the remark this answers is not in the document]
Comment (7alip) [reply apps/dashboard/pages/donations.tsx:44]: `new Date(2023, 01).toISOString() // 2023-01-31T23:00:00.000Z`

**Comment by ertugrul-aydin** [comment]
Ekran Resmi 2023-09-13 14 29 12


Should 'Select Current' button work ? I think I should activate 'on Select' for this. It wasn't in this issue.

File: apps/dashboard/pages/donations.tsx [hunk 5]
```diff
@@ -7,25 +7,39 @@ import { useRouter } from 'next/router'
...
   const donationsQuery = useStrapiRequest<Donation>({
     endpoint: 'donates',
     page: currentPage || 1,
     pageSize: 50,
     filters: {
       ...(searchTerm && { email: { $containsi: searchTerm } }),
+      ...(date && {
+        createdAt: {
+          $gte: startDate,
+          $lte: endDate,
```
Comment (7alip) [inline apps/dashboard/pages/donations.tsx:40]:
```suggestion
          $lt: endDate,
```

`lestThanEqual` will include the first day of the next month.

**Comment by 7alip** [comment]
> Ekran Resmi 2023-09-13 14 29 12
> 
> Should 'Select Current' button work ? I think I should activate 'on Select' for this. It wasn't in this issue.

Actually it's the issue of `MonthPicker` component itself. You don't need to fix in this task. We can create a new task to improve MonthPicker.

**Review by 7alip — APPROVED** [review]
Thank you.

I'm aware of some bugs with MonthPicker but we can fix it in a separate task.

You can merge it **after changing `lte` to `lt`**

**Pull request merged**